packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -122,33 +109,33 @@ class AuthorizationCredentialPassword {
 
 // ignore_for_file: avoid_as, public_member_api_docs
 AuthorizationCredentialAppleID parseAuthorizationCredentialAppleID(
-  Map<dynamic, dynamic> response,
+  Map<dynamic, dynamic>? response,
 ) {
-  if (response['type'] == 'appleid') {
+  if (response?['type'] == 'appleid') {
     return AuthorizationCredentialAppleID(
-      userIdentifier: response['userIdentifier'] as String,
-      givenName: response['givenName'] as String,
-      familyName: response['familyName'] as String,
-      email: response['email'] as String,
-      authorizationCode: response['authorizationCode'] as String,
-      identityToken: response['identityToken'] as String,
-      state: response['state'] as String,
+      userIdentifier: response!['userIdentifier'] as String?,
+      givenName: response['givenName'] as String?,
+      familyName: response['familyName'] as String?,
+      email: response['email'] as String?,
+      authorizationCode: response['authorizationCode'] as String?,
+      identityToken: response['identityToken'] as String?,
+      state: response['state'] as String?,
     );
   } else {
-    throw Exception('Unsupported result type ${response['type']}');
+    throw Exception('Unsupported result type ${response?['type']}');
   }
 }
 
 AuthorizationCredentialPassword parseAuthorizationCredentialPassword(
-  Map<dynamic, dynamic> response,
+  Map<dynamic, dynamic>? response,
 ) {
-  if (response['type'] == 'password') {
+  if (response?['type'] == 'password') {
     return AuthorizationCredentialPassword(
-      username: response['username'] as String,
+      username: response!['username'] as String,
```

Comment:
Why do we only need an `!` here and not in the line below?

Reply:
Because after this line (the first `!`), Dart knows that response must be non-`null` if execution continued :)